src/pages/RFOX/components/Unstake/UnstakeInput.tsx

```diff
@@ -26,143 +49,449 @@ const formControlProps = {
+  )
+
+  const stakingAssetAccountAddress = useMemo(
+    () => (stakingAssetAccountId ? fromAccountId(stakingAssetAccountId).account : undefined),
+    [stakingAssetAccountId],
+  )
+
+  const stakingAssetAccountNumberFilter = useMemo(() => {
+    return {
+      assetId: stakingAssetId,
+      accountId: stakingAssetAccountId,
+    }
+  }, [stakingAssetAccountId, stakingAssetId])
+  const stakingAssetAccountNumber = useAppSelector(state =>
+    selectAccountNumberByAccountId(state, stakingAssetAccountNumberFilter),
+  )
+
+  const feeAssetMarketData = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, feeAsset?.assetId ?? ''),
+  )
+  const stakingAssetMarketData = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, stakingAsset?.assetId ?? ''),
+  )
+
+  const {
+    data: userStakingBalanceOfCryptoBaseUnit,
+    isSuccess: isUserBalanceStakingBalanceOfCryptoBaseUnitSuccess,
+  } = useReadContract({
+    abi: foxStakingV1Abi,
+    address: RFOX_PROXY_CONTRACT_ADDRESS,
+    functionName: 'stakingInfo',
+    args: [stakingAssetAccountAddress ? getAddress(stakingAssetAccountAddress) : ('' as Address)], // actually defined, see enabled below
+    chainId: arbitrum.id,
+    query: {
+      enabled: Boolean(stakingAssetAccountAddress),
+      select: ([stakingBalance]) => stakingBalance.toString(),
+    },
+  })
+
+  const userStakingBalanceCryptoPrecision = useMemo(() => {
+    if (!(userStakingBalanceOfCryptoBaseUnit && stakingAsset)) return
+    return fromBaseUnit(userStakingBalanceOfCryptoBaseUnit, stakingAsset?.precision)
+  }, [stakingAsset, userStakingBalanceOfCryptoBaseUnit])
+
+  const userStakingBalanceUserCurrency = useMemo(() => {
+    if (!(userStakingBalanceCryptoPrecision && stakingAssetMarketData)) return
+
+    return bnOrZero(userStakingBalanceCryptoPrecision)
+      .times(stakingAssetMarketData?.price ?? 0)
+      .toFixed(2)
+  }, [stakingAssetMarketData, userStakingBalanceCryptoPrecision])
+
+  const feeAssetBalanceFilter = useMemo(
+    () => ({
+      accountId: stakingAssetAccountId ?? '',
+      assetId: feeAsset?.assetId,
+    }),
+    [feeAsset?.assetId, stakingAssetAccountId],
+  )
+
+  const feeAssetBalanceCryptoPrecision = useAppSelector(state =>
+    selectPortfolioCryptoPrecisionBalanceByFilter(state, feeAssetBalanceFilter),
+  )
 
   const handleAccountIdChange = useCallback(() => {}, [])
 
-  // @TODO: Need to add a fiat toggle for the input field
-  const hasEnteredValue = useMemo(() => sliderValue > 0, [sliderValue])
+  const hasEnteredValue = useMemo(
+    () => bnOrZero(amountCryptoPrecision).gt(0),
+    [amountCryptoPrecision],
+  )
+
+  const [isFiat, handleToggleIsFiat] = useToggle(false)
 
-  const handleChange = useCallback((value: string, isFiat?: boolean) => {
+  const handleChange = useCallback((_value: string, isFiat?: boolean) => {
     if (isFiat) {
-      setFiatAmount(value)
+      // setFiatAmount(value)
```

Comment:
Is this intentional?

Reply:
It is - we don't need this anymore! Cleaned it up in 6481fe5df6